crates/ruff_python_parser/src/typing.rs

```diff
@@ -93,7 +93,21 @@ fn parse_simple_type_annotation(
 
 fn parse_complex_type_annotation(string_expr: &ExprStringLiteral) -> AnnotationParseResult {
     let mut parsed = parse_expression(string_expr.value.to_str())?;
-    relocate_expr(parsed.expr_mut(), string_expr.range());
+    // Remove quotes from nested forward reference type annotations
+    let range_excluding_quotes =
+        if let Expr::StringLiteral(ExprStringLiteral { ref value, .. }) = *parsed.syntax.body {
+            if let [sl] = value.as_slice() {
```

Comment:
Does this mean we will still get an overflow for this implicitly concatenated example?

```python
def f(x: "'in''\x84'"): 
    pass
```

Reply:
Yes, this case is not handled yet by this PR.

We can see it's valid syntax:

```python
from typing import get_type_hints


def f(x: "'in''\x74'"): 
    pass


print(get_type_hints(f))   # {'x': <class 'int'>}
```

Reply:
I'll add support for those.